src/lib/base64.c

```diff
@@ -46,35 +46,36 @@ static const unsigned char b64dec[256] = {
 void base64_encode(const void *src, size_t src_size, buffer_t *dest)
 {
 	const unsigned char *src_c = src;
-	unsigned char tmp[4];
-	size_t src_pos;
-
-	for (src_pos = 0; src_pos < src_size; ) {
-		tmp[0] = b64enc[src_c[src_pos] >> 2];
-		switch (src_size - src_pos) {
-		case 1:
-			tmp[1] = b64enc[(src_c[src_pos] & 0x03) << 4];
-			tmp[2] = '=';
-			tmp[3] = '=';
-			src_pos++;
-			break;
-		case 2:
-			tmp[1] = b64enc[((src_c[src_pos] & 0x03) << 4) |
-					(src_c[src_pos+1] >> 4)];
-			tmp[2] = b64enc[((src_c[src_pos+1] & 0x0f) << 2)];
-			tmp[3] = '=';
-			src_pos += 2;
-			break;
-		default:
-			tmp[1] = b64enc[((src_c[src_pos] & 0x03) << 4) |
-					(src_c[src_pos+1] >> 4)];
-			tmp[2] = b64enc[((src_c[src_pos+1] & 0x0f) << 2) |
-					((src_c[src_pos+2] & 0xc0) >> 6)];
-			tmp[3] = b64enc[src_c[src_pos+2] & 0x3f];
-			src_pos += 3;
-			break;
-		}
-		buffer_append(dest, tmp, 4);
+	const size_t res_size = (src_size / 3 + (src_size % 3 > 0)) * 4;
```

Comment:
There is a macro for this calculation - `MAX_BASE64_ENCODED_SIZE`.  (Yes, I noticed that it sometimes overestimates by 4 bytes.  I made note of it to fix that up.)

Reply:
Yes, I know about the macro. Should I fix it in this pull request?

Reply:
Up to you if you want to fix the macro.  If you do, it should be a separate commit.